Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/POMNavPage.xaml.cs

```diff
@@ -142,10 +240,125 @@ private void SetElementsGridView()
+            view.GridColsView.Add(new GridColView() { Field = "", Header = "Highlight", WidthWeight = 10, AllowSorting = true, StyleType = GridColView.eGridColStyleType.Template, CellTemplate = (DataTemplate)this.PageGrid.Resources["xHighlightButtonTemplate"] });
+            //view.GridColsView.Add(new GridColView() { Field = nameof(ElementInfo.IsAutoLearned), Header = "Auto Learned", WidthWeight = 10, MaxWidth = 100, AllowSorting = true, ReadOnly = true });
             xMainElementsGrid.SetAllColumnsDefaultView(view);
             xMainElementsGrid.InitViewItems();
             xMainElementsGrid.ChangeGridView(eGridView.RegularView.ToString());
+
+            xMainElementsGrid.AddToolbarTool(eImageType.GoBack, "Add to Actions", new RoutedEventHandler(AddFromPOMNavPage));
+        }
+
+        private void AddFromPOMNavPage(object sender, RoutedEventArgs e)
+        {
+            if (xMainElementsGrid.Grid.SelectedItems != null && xMainElementsGrid.Grid.SelectedItems.Count > 0)
+            {
+                foreach (ElementInfo elemInfo in xMainElementsGrid.Grid.SelectedItems)
+                {
+                    Act instance = GenerateRelatedAction(elemInfo);
+                    if(instance != null)
+                    {
+                        instance.Active = true;
+
+                    }
+                    mContext.BusinessFlow.AddAct(instance, true);
+                }
+
+                int selectedActIndex = -1;
+                ObservableList<IAct> actsList = mContext.BusinessFlow.CurrentActivity.Acts;
+                if (actsList.CurrentItem != null)
+                {
+                    selectedActIndex = actsList.IndexOf((Act)actsList.CurrentItem);
+                }
+                if (selectedActIndex >= 0)
+                {
+                    actsList.Move(actsList.Count - 1, selectedActIndex + 1);
+
+                }
+            }
+            else
+                Reporter.ToUser(eUserMsgKey.NoItemWasSelected);
+        }
+
+        private static Act GenerateRelatedAction(ElementInfo elementInfo)
+        {
+            Act instance;
+            IPlatformInfo mPlatform = PlatformInfoBase.GetPlatformImpl(ePlatformType.Web);
+            ElementActionCongifuration actionConfigurations = new ElementActionCongifuration
+            {
+                Description = "UIElement Action : " + elementInfo.ItemName,
+                Operation = ActUIElement.eElementAction.NotExist.ToString(),
+                LocateBy = eLocateBy.POMElement,
+                LocateValue = elementInfo.ParentGuid.ToString() + "_" + elementInfo.Guid.ToString(),
+                ElementValue = "",
+                AddPOMToAction = true,
+                POMGuid = elementInfo.ParentGuid.ToString(),
+                ElementGuid = elementInfo.Guid.ToString(),
+                LearnedElementInfo = elementInfo,
+            };
+
+            switch (elementInfo.ElementTypeEnum)
```

Comment:
you not supposed to set the operation of the action in this page, all need to be done by mPlatform.GetPlatformAction() in generic code, what if tomorrow we will have POM of Java application and we will want to add action from one of it elements? so we will do switch case and check if element is Div? no, this is why the action setup needs to be done by the proper platform (web, java, etc.)

Reply:
Okay Done. Updated WebPlatforms's GetPlatformActionByElementInfo with the same.